multilinear_extensions/src/mle.rs

```diff
@@ -753,239 +609,291 @@ impl<E: ExtensionField> MultilinearExtension<E> for DenseMultilinearExtension<E>
+    pub fn get_base_field_vec(&self) -> &[E::BaseField] {
+        match &self.evaluations() {
+            FieldType::Base(evaluations) => evaluations.as_ref(),
+            _ => panic!("evaluation not in base field"),
         }
     }
-}
 
-impl<'a, E: ExtensionField> MultilinearExtension<E> for RangedMultilinearExtension<'a, E> {
-    type Output = DenseMultilinearExtension<E>;
-    fn fix_variables(&self, partial_point: &[E]) -> Self::Output {
-        // TODO: return error.
-        assert!(
-            partial_point.len() <= self.num_vars(),
-            "invalid size of partial point"
+    /// splits the MLE into `num_chunks` parts, where each part contains disjoint mutable pointers
+    /// to the original data (either borrowed mutably or owned).
+    ///
+    /// # panics
+    /// panics if:
+    /// - `num_chunks` is zero,
+    /// - the data length is not divisible by `num_chunks`,
+    /// - or the underlying data is neither owned nor mutably borrowed.
+    pub fn split_mle_into_chunks(
+        &'a mut self,
+        num_chunks: usize,
+    ) -> Vec<MultilinearExtension<'a, E>> {
+        assert!(num_chunks > 0, "num_chunks must be > 0");
+        let len = self.evaluations.len();
+        assert_eq!(
+            len % num_chunks,
+            0,
+            "Evaluation length must be divisible by num_chunks"
         );
+        let chunk_size = len / num_chunks;
+        let num_vars_per_chunk = self.num_vars - ceil_log2(num_chunks);
 
-        if !partial_point.is_empty() {
-            let first = partial_point[0];
-            let inner = self.inner;
-            let mut mle = op_mle!(inner, |evaluations| {
-                DenseMultilinearExtension::from_evaluations_ext_vec(
-                    self.num_vars() - 1,
-                    // syntax: evaluations[start..(start+offset)]
-                    evaluations[self.start..][..self.offset]
-                        .chunks(2)
-                        .map(|buf| first * (buf[1] - buf[0]) + buf[0])
-                        .collect(),
-                )
-            });
-            mle.fix_variables_in_place(&partial_point[1..]);
-            mle
-        } else {
-            self.inner.clone()
-        }
-    }
+        match &mut self.evaluations {
+            // handle splitting a mutable borrowed slice into disjoint mutable slices
+            FieldType::Base(SmartSlice::BorrowedMut(slice)) => {
+                let mut result = Vec::with_capacity(num_chunks);
+                let ptr = slice.as_mut_ptr();
+                for i in 0..num_chunks {
+                    let start = i * chunk_size;
+                    let chunk =
+                        unsafe { std::slice::from_raw_parts_mut(ptr.add(start), chunk_size) };
+                    result.push(MultilinearExtension {
+                        evaluations: FieldType::Base(SmartSlice::BorrowedMut(chunk)),
+                        num_vars: num_vars_per_chunk,
+                    });
+                }
```

Comment:
It's more readable if we use 
```rust
slice
    .chunks_mut(chunk_size)
    .map(|chunk| MultilinearExtension {
        evaluations: FieldType::Base(SmartSlice::BorrowedMut(chunk)),
        num_vars: num_vars_per_chunk,
    })
    .collect::<Vec<_>>()
```

Reply:
updated this part into function style https://github.com/scroll-tech/ceno/pull/940/commits/508843d1caf27bc427dc94b930cf767156f9d863

Reply:
why do we bother to use unsafe function instead of just `.chunks_mut()`?

Reply:
You are right 👍  I didnt realized in first place that chunk also can be convert to mutable slice freely
updated https://github.com/scroll-tech/ceno/pull/940/commits/fc733ba4520165242730b46e003c25b93d69a1d0